mm2src/mm2_main/src/lp_swap/swap_watcher.rs

```diff
@@ -368,7 +369,12 @@ impl State for SpendMakerPayment {
     async fn on_changed(self: Box<Self>, watcher_ctx: &mut WatcherContext) -> StateResult<Self::Ctx, Self::Result> {
         let spend_fut = watcher_ctx
             .maker_coin
-            .send_maker_payment_spend_preimage(&watcher_ctx.data.maker_payment_spend_preimage, &self.secret.0);
+            .send_maker_payment_spend_preimage(SendMakerPaymentSpendPreimageInput {
+                preimage: &watcher_ctx.data.maker_payment_spend_preimage,
+                secret: &self.secret.0,
+                secret_hash: &watcher_ctx.data.secret_hash,
+                taker_pub: &watcher_ctx.verified_pub,
+            });
```

Comment:
Could you create an instance of `SendMakerPaymentSpendPreimageInput` before the method call? It will take less lines.
```rust
let args = SendMakerPaymentSpendPreimageInput {
            preimage: &watcher_ctx.data.maker_payment_spend_preimage,
            secret: &self.secret.0,
            secret_hash: &watcher_ctx.data.secret_hash,
            taker_pub: &watcher_ctx.verified_pub,
        };
let spend_fut = watcher_ctx.maker_coin.send_maker_payment_spend_preimage(args);```

Reply:
Yes it takes one line less if I use a separate variable (if the variable name is short, otherwise it takes more), but I think the code is slightly easier to read and understand the way it is now, so I'd prefer to use the struct directly as parameter.